src/client/driver/driver.js

```diff
@@ -732,7 +936,47 @@ export default class Driver {
 
         this.contextStorage.setItem(this.COMMAND_EXECUTING_FLAG, false);
         this.contextStorage.setItem(this.EXECUTING_IN_IFRAME_FLAG, false);
+        this.contextStorage.setItem(this.PENDING_WINDOW_SWITCHING_FLAG, false);
 
         this._onReady(status);
     }
+
+    _initParentWindowLink () {
+        this.parentWindowDriverLink = new ParentWindowDriverLink(window);
+    }
+
+    _getDriverRole () {
```

Comment:
can we rename this method somehow?
something like `isMasterPage`?

Reply:
At present, driver has a role: master or replica. see the [PR description](https://github.com/DevExpress/testcafe/pull/4485#issue-340920020)  I will rewrite this code as follows:
```js
this._getDriverRole()
            .then(role=> {
                if (role === DriverRole.Master)
                    this._startInternal();
                else
                    this._initParentWindowLink();
            });
```